src/utils/moderation.rs

```diff
@@ -72,17 +72,25 @@ pub async fn mute_on_join(ctx: &serenity::Context, new_member: &mut serenity::Me
 
 pub fn queue_unmute_member(
     http: &Arc<Http>,
+    cache: &Arc<Cache>,
     database: &Arc<Database>,
-    member: &Member,
+    guild_id: serenity::GuildId,
+    user_id: serenity::UserId,
     mute_role_id: u64,
     mute_duration: u64,
 ) -> JoinHandle<Option<Error>> {
+    let cache = cache.clone();
     let http = http.clone();
     let database = database.clone();
-    let mut member = member.clone();
+    // let mut member: Member = panic!("explod");
+    // let mut member = member.clone();
 
     tokio::spawn(async move {
         tokio::time::sleep(std::time::Duration::from_secs(mute_duration)).await;
+        let Some(mut member) = cache.member(guild_id, user_id) else {
+            // explod. they are not member lol
```

Comment:
Instead of spawning a thread, then waiting and then dismissing the assumption that the user is a member, this should be done before spawning a thread.
```suggestion
```

Reply:
We can't. The member might have left after the mute expires so we cannot really do it before we spawn the thread.

You did make me realize that the let-else was placed too early, but that has since been fixed.